### PR TITLE
netcdf: add netcdf/4.7.4 recipe

### PR DESCRIPTION
**Pull request by madebr**
Specify library name and version:  **netcdf/4.7.4**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/netcdf/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,125 @@
+from conans import CMake, ConanFile, tools
+import glob
+import os
+
+
+class NetcdfConan(ConanFile):
+    name = "netcdf"
+    description = "The Unidata network Common Data Form (netCDF) is an interface for scientific data access and a freely-distributed software library that provides an implementation of the interface."
+    topics = "UNKNOWN TOPICS"
+    license = "https://github.com/Unidata/netcdf-c"
```
Comment (uilianries) [inline recipes/netcdf/all/conanfile.py:10]:
```suggestion
    license = "BSD-3-Clause"
```

File: recipes/netcdf/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,125 @@
+from conans import CMake, ConanFile, tools
+import glob
+import os
+
+
+class NetcdfConan(ConanFile):
+    name = "netcdf"
+    description = "The Unidata network Common Data Form (netCDF) is an interface for scientific data access and a freely-distributed software library that provides an implementation of the interface."
+    topics = "UNKNOWN TOPICS"
```
Comment (uilianries) [inline recipes/netcdf/all/conanfile.py:9]:
```suggestion
    topics = ("unidata", "unidata-netcdf", "networking")
```

File: recipes/netcdf/all/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+        if self._with_hdf5:
+            self.requires("hdf5/1.12.0")
+        if self.options.dap:
+            self.requires("libcurl/7.73.0")
+
+    def build_requirements(self):
+        if tools.os_info.is_windows and not tools.get_env("CONAN_BASH_PATH"):
+            self.build_requires("msys2/20200517")
+        if self.settings.compiler == "Visual Studio":
+            self.build_requires("automake/1.16.2")
```
Comment (prince-chrismc) [inline recipes/netcdf/all/conanfile.py:67]: If this is build by CMake are these required?

File: recipes/netcdf/all/conanfile.py [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+
+    def source(self):
```
Comment (madebr) [inline recipes/netcdf/all/conanfile.py:69]:
```suggestion
            self.requires("libcurl/7.73.0")

    def source(self):
```

File: recipes/netcdf/all/conanfile.py (hunk 3 above)
Comment (madebr) [reply recipes/netcdf/all/conanfile.py:67]: No, remainder of my conan template

**Review by prince-chrismc — APPROVED** [review]

**Comment by prince-chrismc** [comment]
Please retrigger CI =)

**Comment by madebr** [comment]
@prince-chrismc 
Thanks for the reminder

**Review by uilianries — APPROVED** [review]
LGTM

File: recipes/netcdf/all/conanfile.py [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "netcdf4": True,
+        "with_hdf5": True,
+        "cdf5": True,
+        "dap": True,
+    }
+    generators = "cmake_find_package", "cmake_find_package_multi", "cmake"
```
Comment (SSE4) [inline recipes/netcdf/all/conanfile.py:31]: I rarely see both `cmake_find_package` and `cmake_find_package_multi` are needed

File: recipes/netcdf/all/patches/0001-cmake-subproject.patch [hunk 6]
```diff
@@ -0,0 +1,157 @@
...
++++ cmake/modules/FindMakeDist.cmake
+@@ -153,9 +153,9 @@ macro(enable_makedist)
+     get_source_file_property(__makedist_extradist   "${__makedist_source}" EXTRADIST)
+ 
+     if(NOT __makedist_generated)
+-      file(RELATIVE_PATH __makedist_cur_path "${CMAKE_SOURCE_DIR}" "${__makedist_full_cur_path}")
++      file(RELATIVE_PATH __makedist_cur_path "${PROJECT_SOURCE_DIR}" "${__makedist_full_cur_path}")
+     elseif(__makedist_extradist)
+-      file(RELATIVE_PATH __makedist_cur_path "${CMAKE_BINARY_DIR}" "${__makedist_full_cur_path}")
++      file(RELATIVE_PATH __makedist_cur_path "${PROJCT_BINARY_DIR}" "${__makedist_full_cur_path}")
```
Comment (SSE4) [inline recipes/netcdf/all/patches/0001-cmake-subproject.patch:94]:
```suggestion
+      file(RELATIVE_PATH __makedist_cur_path "${PROJECT_BINARY_DIR}" "${__makedist_full_cur_path}")
```

**Review by SSE4 — APPROVED** [review]

**Comment by Croydon** [comment]
Oh damn. While going through issues I noticed this comment: https://github.com/conan-io/conan-center-index/issues/1850

This probably should have been named netcdf-c instead. netcdf-cxx also exists.

**Comment by d70-t** [comment]
Thank you all for finishing the netcdf recipe! I got sidetracked with my PR and am very happy that I don't have to worry about it anymore. :+1: :100:

**Pull request merged**